zenoh/src/api/encoding.rs

```diff
@@ -582,21 +466,8 @@ impl Encoding {
 
     const ID_TO_STR: phf::Map<EncodingId, &'static str> = phf_map! {
         0u16 => "zenoh/bytes",
-        1u16 => "zenoh/int8",
-        2u16 => "zenoh/int16",
-        3u16 => "zenoh/int32",
-        4u16 => "zenoh/int64",
-        5u16 => "zenoh/int128",
-        6u16 => "zenoh/uint8",
-        7u16 => "zenoh/uint16",
-        8u16 => "zenoh/uint32",
-        9u16 => "zenoh/uint64",
-        10u16 => "zenoh/uint128",
-        11u16 => "zenoh/float32",
-        12u16 => "zenoh/float64",
-        13u16 => "zenoh/bool",
-        14u16 => "zenoh/string",
-        15u16 => "zenoh/error",
+        1u16 => "zenoh/string",
+        15u16 => "zenoh/serialized",
```

Comment:
You really want to "reserve" encoding slots?

Reply:
I think this makes sense. Later we can even restore primitive types encoding to these slots, with names like `zenoh/serialized;u8`, `zenoh/serialized;i16`, etc - when the schema for zenoh encoding is elaborated (this is @kydos 's idea)

Reply:
Then you will surely need more than 2..16 range, so why not reserving more? Why do we care having zenoh encodings first? (except for `zenoh/bytes` which must obviously be 0)

Reply:
Mostly because it allows us not to change existing numeration. Yes, it's internal and as we know these codes are not used anywhere, but as it costs nothing to us, better not to break compatibility when we can.
But this is really minor question. If you are strongly against this gap, let's remove it, it's cheaper than discuss it too long.

Reply:
@DenisBiryukov91 is also against the gap and this makes zenoh-pico a bit more complex. Then let's renumber them, fixing

Reply:
renumbered